src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/RSA.cs

```diff
@@ -49,29 +48,97 @@ public static RSA Create(RSAParameters parameters)
-        public virtual byte[] SignHash(byte[] hash, HashAlgorithmName hashAlgorithm, RSASignaturePadding padding) { throw DerivedClassMustOverride(); }
-        public virtual bool VerifyHash(byte[] hash, byte[] signature, HashAlgorithmName hashAlgorithm, RSASignaturePadding padding) { throw DerivedClassMustOverride(); }
+        public virtual byte[] Encrypt(byte[] data, RSAEncryptionPadding padding) => throw DerivedClassMustOverride();
+        public virtual byte[] Decrypt(byte[] data, RSAEncryptionPadding padding) => throw DerivedClassMustOverride();
+        public virtual byte[] SignHash(byte[] hash, HashAlgorithmName hashAlgorithm, RSASignaturePadding padding) => throw DerivedClassMustOverride();
+        public virtual bool VerifyHash(byte[] hash, byte[] signature, HashAlgorithmName hashAlgorithm, RSASignaturePadding padding) => throw DerivedClassMustOverride();
 
-        protected virtual byte[] HashData(byte[] data, int offset, int count, HashAlgorithmName hashAlgorithm) { throw DerivedClassMustOverride(); }
-        protected virtual byte[] HashData(Stream data, HashAlgorithmName hashAlgorithm) { throw DerivedClassMustOverride(); }
+        protected virtual byte[] HashData(byte[] data, int offset, int count, HashAlgorithmName hashAlgorithm) => throw DerivedClassMustOverride();
+        protected virtual byte[] HashData(Stream data, HashAlgorithmName hashAlgorithm) => throw DerivedClassMustOverride();
 
-        private static Exception DerivedClassMustOverride()
+        public virtual bool TryDecrypt(ReadOnlySpan<byte> source, Span<byte> destination, RSAEncryptionPadding padding, out int bytesWritten)
         {
-            return new NotImplementedException(SR.NotSupported_SubclassOverride);
+            byte[] result = Decrypt(source.ToArray(), padding);
+
+            if (destination.Length >= result.Length)
+            {
+                new ReadOnlySpan<byte>(result).CopyTo(destination);
+                bytesWritten = result.Length;
+                return true;
+            }
+
+            bytesWritten = 0;
+            return false;
         }
 
-        public virtual byte[] DecryptValue(byte[] rgb) 
+        public virtual bool TryEncrypt(ReadOnlySpan<byte> source, Span<byte> destination, RSAEncryptionPadding padding, out int bytesWritten)
         {
-            throw new NotSupportedException(SR.NotSupported_Method); // Same as Desktop
+            byte[] result = Encrypt(source.ToArray(), padding);
+
+            if (destination.Length >= result.Length)
+            {
+                new ReadOnlySpan<byte>(result).CopyTo(destination);
+                bytesWritten = result.Length;
+                return true;
+            }
+
+            bytesWritten = 0;
+            return false;
         }
 
-        public virtual byte[] EncryptValue(byte[] rgb) 
+        protected virtual bool TryHashData(ReadOnlySpan<byte> source, Span<byte> destination, HashAlgorithmName hashAlgorithm, out int bytesWritten)
```

Comment:
There should probably be a dedicated test for these in the Algorithms.Tests library, rather than getting an implicit "well, RSACryptoServiceProvider on Windows works", so that we have coverage in the event we upgrade that one in the future.